### ISSUE TITLE
Error creating a qlist from CLUMPP files

### ISSUE THREAD

**Issue by iviceron** [royfrancis/pophelper#12]
Thanks for this tool! Very useful.
I used Pophelper a year ago and now I am learning how to use the latest version 2.2.2. Unfortunately, I am having an error when I tried to create a qlist from my CLUMMP files. I also tried with your example file and I got the same error.  

This is the error using your example file:
```r
#CLUMPP file
tabs3<-system.file("STRUCTUREpop_K4-combined-merged.txt", package="pophelper")
#create a qlist
readQ(tabs3, filetype = 'clumpp') 

Error in read.table(files[i], header = FALSE, sep = seps[k], nrows = 1,  : 
  no lines available in input
In addition: Warning message:
In file(con, "r") :
  file("") only supports open = "w+" and open = "w+b": using the former
Error in read.table(files[i], header = FALSE, sep = seps[k], nrows = 1,  : 
  no lines available in input
Error in read.table(files[i], header = FALSE, sep = seps[k], nrows = 1,  : 
  no lines available in input
Error in pophelper:::readQClumpp(files[i]) : 
  readQClumpp: Incorrect input format.
```

I appreciate your input to understand what is the problem.

Thanks!

**Comment by royfrancis** [comment]
In your example, the `system.file()` path is wrong which returns an empty string to `tabs3`. And the empty string gives an error in the next line with `readQ()`.

Adding `files/` makes it work for me.
```r
library(pophelper)
tabs3 <- system.file("files/STRUCTUREpop_K4-combined-merged.txt", package="pophelper")
readQ(tabs3)
```

But, that doesn't explain why it doesn't work with your files. Are your CLUMPP files output from pophelper or CLUMPP? Have you modified them in any way? Feel free to send me a sample file if you still have trouble getting it to work.

**Comment by iviceron** [comment]
Dear Roy!
Thanks for your quick reply.  I added files/, but still, it doesn't work. Same error.
My files come from the software BAPS (http://www.helsinki.fi/bsg/software/BAPS/). Since Pophelper doesn't accept yet the BAPS output files, I made it by hand as a CLUMPP files, because is the closest format. 

I appreciate you take look at my files.  Attached are three files as an example, the original output file from BAPS, my CLUMPP file made by hand from the output BAPS file and the metadata file.

Thanks in advance for your time!

[virosu_m3n3_47_23484_k2_asCLUMP2.txt](https://github.com/royfrancis/pophelper/files/1321156/virosu_m3n3_47_23484_k2_asCLUMP2.txt)
[virosu_m3n3_47_23484_BAPS_labels.txt](https://github.com/royfrancis/pophelper/files/1321159/virosu_m3n3_47_23484_BAPS_labels.txt)
[virosu_m3n3_23484_47_pop_from_grad_admixtured.mat.txt](https://github.com/royfrancis/pophelper/files/1321163/virosu_m3n3_23484_47_pop_from_grad_admixtured.mat.txt)

**Comment by royfrancis** [comment]
yes, your CLUMPP formatted file looks fine. It should work. Are you running it like below?
`readQ("virosu_m3n3_47_23484_k2_asCLUMP2.txt")`

Above should work if the file is in the current working directory else a relative or full path must be provided.
`readQ("./bapsfiles/virosu_m3n3_47_23484_k2_asCLUMP2.txt")`

**Comment by iviceron** [comment]
Hi Roy!
Thanks for your help. I use readQ("./bapsfiles/virosu_m3n3_47_23484_k2_asCLUMP2.txt") and it works without problem! 

However, when I tried to import my metadata, I got again an error :
label1 <-read.delim(system.file("./bapsfiles/virosu_m3n3_47_23484_labels.txt", package="pophelper"), header=F,stringsAsFactors=F)

Error in read.table(file = file, header = header, sep = sep, quote = quote,  : 
  no lines available in input
In addition: Warning message:
In file(file, "rt") :
  file("") only supports open = "w+" and open = "w+b": using the former

What do you think is happening?

Thanks!

**Comment by royfrancis** [comment]
Ok. I see the problem. The `system.file()` function is used to read stuff from within the R package. The examples use `system.file()` because the file to be read in is inside the R package. Do not use it to read your own local files. To read in your own files just use:

readQ("yourfile.txt")

The above works only if the file `yourfile.txt` is in the current working directory. If not, then set the working directory to the location of the file. Or, provide a path to the file. 

In the next release, I can add a function to read in BAPS run files, so you don't have to modify them. Would you be able to share a set of BAPS output run files (for various K)? And also the BAPS version you use? Thanks.

**Comment by iviceron** [comment]
Hi Roy!
Thanks a lot for your time solving my problem. 

I am glad you consider to include BAPS outputs for your new version. Attached are some output files from BAPS. I used the latest version of BAPS ver. 6.0.

Be aware that while Structure analyzes the best K and the level of admixture per individual of each K simultaneously; in BAPS, those two analyzes are separated and consecutive. First, you determine the best structure and then, only if the best K is higher than 1, you then do the second part of the analysis which is to define the level of the admixture of each individual. Therefore, the BAPS output for the first part (i.e., what is the best structure) doesn't have results for each K as Structure. It is just one output file that gives you: 1) the best K with its probability (based on the Ks you tested and their replicates), and 2) a summary of what individuals belong to each one of the clusters.  

If the best K=1, you analysis finish here. In that case, for Pophelper, I guess you have to do the figure by hand. But, if you best K is higher than 1, you can run the admixture analysis. The output file in those cases has a very similar to Structure and CLUMPP format (i.e., a list of individuals with a value of Q for each cluster). So, I guess this is the file you need to work for the new version of Pophelper. 

I am attached one example where best K=1 (one file, the mixture, vochfe) and one example when best K is higher than 1 (two files, the mixture, and then, the admixture analysis, virosu). For those examples, I use as priors the genetic and the spatial information of each sampling locality. 

If you have more questions about BAPS, let me know. Thanks again for your quick response to solve my problems using Pophelper.

[virosu_m3n3_23484_47_pop_from_grad_admixtured.mat.txt](https://github.com/royfrancis/pophelper/files/1337327/virosu_m3n3_23484_47_pop_from_grad_admixtured.mat.txt)
[vochfe_m3n1_81_4565_grad_mixed.mat.txt](https://github.com/royfrancis/pophelper/files/1337335/vochfe_m3n1_81_4565_grad_mixed.mat.txt)

[virosu_m3n3_23484_47_pop_from_grad_mixtured.mat.txt](https://github.com/royfrancis/pophelper/files/1337325/virosu_m3n3_23484_47_pop_from_grad_mixtured.mat.txt)

**Comment by royfrancis** [comment]
BAPS run files can be read in using `readQ()`. Fixed in version `2.2.4`.
